src/database.rs

```diff
@@ -76,41 +98,90 @@ pub fn get_data_from_db(
             .collect();
 
         if missing_sgs.is_empty() {
-            println!("No missing state groups");
+            // println!("No missing state groups");
             break;
         }
 
         missing_sgs.sort_unstable();
         missing_sgs.dedup();
 
-        println!("Missing {} state groups", missing_sgs.len());
+        // println!("Missing {} state groups", missing_sgs.len());
 
-        let map = get_missing_from_db(&mut client, &missing_sgs);
-        state_group_map.extend(map.into_iter());
+        // find state groups not picked up already and add them to the map
+        let map = get_missing_from_db(&mut client, &missing_sgs, min_state_group, max_group_found);
+        for (k, v) in map.into_iter() {
+            state_group_map.entry(k).or_insert(v);
+        }
```

Comment:
Why this change? Also `.into_iter()` is implicit in for loops, so you can do that as `for (k, v) in map { .. }`

Reply:
I did this so that state groups you get from get_missing_from_db cannot overwrite the entries in state_group_map, which they could before. 

Unless something goes wrong, that should never happen atm. so it was just for peace of mind when I was editing get_missing_from_db (I guess it could be a panic if it was trying to replace anything? As would mean compressor potentially unsafe)

Reply:
Ah, fair.